nexus/db-queries/src/db/datastore/deployment.rs

```diff
@@ -3036,93 +3036,24 @@ mod tests {
-            query_count!(bp_omicron_zone, blueprint_id),
-            query_count!(bp_omicron_zone_nic, blueprint_id),
-            query_count!(bp_clickhouse_cluster_config, blueprint_id),
-            query_count!(bp_clickhouse_keeper_zone_id_to_node_id, blueprint_id),
-            query_count!(bp_clickhouse_server_zone_id_to_node_id, blueprint_id),
-            query_count!(bp_oximeter_read_policy, blueprint_id),
-            query_count!(bp_pending_mgs_update_sp, blueprint_id),
-            query_count!(bp_pending_mgs_update_rot, blueprint_id),
-            query_count!(bp_pending_mgs_update_rot_bootloader, blueprint_id),
-            query_count!(bp_pending_mgs_update_host_phase_1, blueprint_id),
-        ] {
-            let count: i64 = result.unwrap();
-            assert_eq!(
-                count, 0,
-                "nonzero row count for blueprint \
-                 {blueprint_id} in table {table_name}"
-            );
-            tables_checked.insert(table_name);
-        }
+        let counts = BlueprintTableCounts::new(datastore, blueprint_id).await;
 
-        // Look for likely blueprint-related tables that we didn't check.
-        let mut query = QueryBuilder::new();
-        query.sql(
-            "SELECT table_name \
-            FROM information_schema.tables \
-            WHERE table_name LIKE 'bp\\_%'",
+        // All tables should be empty (no exceptions for deleted blueprints)
+        assert!(
+            counts.all_empty(),
+            "Blueprint {blueprint_id} not fully deleted. Non-empty tables: {:?}",
+            counts.non_empty_tables()
         );
-        let tables_unchecked: Vec<String> = query
-            .query::<diesel::sql_types::Text>()
-            .load_async(&*conn)
-            .await
-            .expect("Failed to query information_schema for tables")
-            .into_iter()
-            .filter(|f: &String| {
-                let t = f.as_str();
-                !tables_ignored.contains(t) && !tables_checked.contains(t)
-            })
-            .collect();
-        if !tables_unchecked.is_empty() {
-            // If you see this message, you probably added a blueprint table
-            // whose name started with `bp_*`.  Add it to the block above so
-            // that this function checks whether deleting a blueprint deletes
-            // rows from that table.  (You may also find you need to update
-            // blueprint_delete() to actually delete said rows.)
-            panic!(
-                "found table(s) that look related to blueprints, but \
-                 aren't covered by ensure_blueprint_fully_deleted(). \
-                 Please add them to that function!\n
-                 Found: {}",
-                tables_unchecked.join(", ")
-            );
+
+        // Verify no new blueprint tables were added without updating this function
+        if let Err(msg) = counts.verify_all_tables_covered(datastore).await {
+            panic!("{}", msg);
```

Comment:
I think now that this is in `BlueprintTableCounts::new()`, we can remove this here.

Reply:
Done, thank you for catching this 🙂